sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -203,30 +219,30 @@ abstract class JdbcDialect extends Serializable with Logging{
   def compileAggregate(aggFunction: AggregateFunc): Option[String] = {
     aggFunction match {
       case min: Min =>
-        val sql = min.column match {
+        min.column match {
```

Comment:
we can do `compileExpression(min.column).map(v => s"MIN($v)")` directly if https://github.com/apache/spark/pull/35494/files#r814549510 is done